dev/deno.ts

```diff
@@ -768,24 +780,17 @@ const denoDoc: Fig.Subcommand = {
           // The names are added to a set because duplicates are common.
           const names = new Set<string>();
 
-          // More nodes should be visible if --private was used. This is two
-          // separate loops because the logic is fundamentally different - one
-          // loop just adds all the nodes, while the other is a filter. It also
-          // avoids checking showPrivateSymbols in each iteration, since that
-          // value won't change.
-          if (showPrivateSymbols) {
-            for (const node of suggestNodes) {
-              names.add(node.name);
-            }
-          } else {
-            for (const node of suggestNodes) {
-              // Deno includes imports in the JSON regardless of privacy,
-              // probably so it's easier to build a module graph. However,
-              // imports are (by definition) private, so they must be removed.
-              if (node.kind === "import") continue;
-
+          // More nodes should be visible if --private was used.
+          for (const node of suggestNodes) {
+            if (showPrivateSymbols) {
```

Comment:
I even prefer refactoring again to join that call.
```suggestion
            if (showPrivateSymbols || node.kind !== "import") {
```

Reply:
I think leaving it as-is makes it much easier to read, even if it's longer. @QuiiBz thoughts?